pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +391,26 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) []map[string]interface{} {
+	registries := make([]map[string]interface{}, len(registryMap))
+	for k, v := range registryMap {
+		registries[k] = v
```

Comment:
Index `k` may be out of bounds if somehow we have request that contains something like `rancherKubernetesEngineConfig.privateRegistries[7].url` when we only have 3 private registries already set on the cluster. I think previous solution was safer/better as we were just essentially adding it as privateRegistries[3] in this case (as we always appended to the list if index didn't exist).

Still, we're talking hypothetical situations here so this solution feels overly complex for fixing the issue that doesn't exist.

Reply:
The out-of-bound issue will not occur because we only process the answers to which questions are in `clusterTemplateRevision.Spec.Questions`:
```
for _, question := range clusterTemplateRevision.Spec.Questions {
```

let's say we have 3 private registries defined in the cluster template, then the questions will be 
```
rancherKubernetesEngineConfig.privateRegistries[0].url
rancherKubernetesEngineConfig.privateRegistries[0].username
rancherKubernetesEngineConfig.privateRegistries[0].password
rancherKubernetesEngineConfig.privateRegistries[1].url
rancherKubernetesEngineConfig.privateRegistries[1].username
rancherKubernetesEngineConfig.privateRegistries[1].password
rancherKubernetesEngineConfig.privateRegistries[2].url
rancherKubernetesEngineConfig.privateRegistries[2].username
rancherKubernetesEngineConfig.privateRegistries[2].password
```
If the request contains `rancherKubernetesEngineConfig.privateRegistries[7].url`, it will be dropped because it is not in the above list. 

However, there is a special case: the cluster template is created in a malformed format where the indexes for private registries are not continuous values, which I think should never happen. But even in this case, the current implementation should still be able to process the answers properly.

Reply:
This is discussed offline and we decide to return an error if the index is not consecutive and less than the count of registries.